### PR TITLE
[MM-37202] Use native upsert query in StatusStore.SaveOrUpdate()

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Replacing the two queries (select + insert) which were causing some errors (seen a lot of these during recent load-tests) with a single upsert query.

#### Ticket

https://mattermost.atlassian.net/browse/MM-37202

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by agnivade — APPROVED** [review]
Ah you are doing our team's work now :)

**Comment by streamer45** [comment]
> Ah you are doing our team's work now :)

Old habits, or maybe planning for the future :roll_eyes:

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

**Comment by streamer45** [comment]
@furqanmlk 

A smoke-test around changing user statuses should be sufficient here.

**Comment by srkgupta** [comment]
Creating Spinwick for my testing purpose. Not related to this PR.

**Review by furqanmlk — APPROVED** [review]
User Status E2E Passed ✅ 

image

image

image

image

image

image

image

File: store/sqlstore/status_store.go [hunk 1]
```diff
@@ -39,18 +38,29 @@ func (s SqlStatusStore) createIndexesIfNotExists() {
...
-			return errors.Wrap(err, "failed to update Status")
-		}
+func (s SqlStatusStore) SaveOrUpdate(st *model.Status) error {
+	query := s.getQueryBuilder().
+		Insert("Status").
+		Columns("UserId", "Status", "Manual", "LastActivityAt", "DNDEndTime", "PrevStatus").
+		Values(st.UserId, st.Status, st.Manual, st.LastActivityAt, st.DNDEndTime, st.PrevStatus)
+
+	if s.DriverName() == model.DatabaseDriverMysql {
+		query = query.SuffixExpr(sq.Expr("ON DUPLICATE KEY UPDATE UserId = ?, Status = ?, Manual = ?, LastActivityAt = ?, DNDEndTime = ?, PrevStatus = ?",
```
Comment (agnivade) [inline store/sqlstore/status_store.go:48]: Sorry, I missed this previously, but do we need to update the userId here and the below? Seems redundant as that's the primary key here.

Comment (streamer45) [reply store/sqlstore/status_store.go:48]: Yes that's redundant given it will be the same value so it can be safely removed.

**Pull request merged**